apollo-federation/src/operation/rebase.rs

```diff
@@ -309,6 +310,41 @@ impl FieldSelection {
     }
 }
 
+impl FragmentSpread {
+    /// - `named_fragments`: named fragment definitions that are rebased for the subgraph.
+    // Note: This method is used during operation optimization.
+    pub(crate) fn rebase_on(
+        &self,
+        parent_type: &CompositeTypeDefinitionPosition,
+        schema: &ValidFederationSchema,
+        named_fragments: &NamedFragments,
+        error_handling: RebaseErrorHandlingOption,
+    ) -> Result<Option<FragmentSpread>, FederationError> {
+        let Some(named_fragment) = named_fragments.get(&self.data().fragment_name) else {
+            return if let RebaseErrorHandlingOption::ThrowError = error_handling {
+                Err(FederationError::internal(format!(
+                    "Cannot rebase {} fragment if it isn't part of the provided fragments",
+                    self.data().fragment_name
+                )))
+            } else {
+                Ok(None)
+            };
+        };
+        debug_assert_eq!(*schema, self.data().schema);
```

Comment:
can you add a message to those debug so it is easier to figure out what is the problem?